### DISCUSSION TITLE
Securely store secrets inside env file

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#5114] [Q&A]
Hi there, I know its not a good idea to store secrets in env file because env can be seen in the source code in production. 
However in Vite it can prevent certain environment variables to be leaked into source code by having a convention like below

![image](https://github.com/web-infra-dev/modern.js/assets/52765572/a2015332-9dd8-4998-9528-2f51c8e02b91)

Does Modern JS have similar thing that prevents env variables to be leaked into source code when compiled?

thanks :)

**Comment by clChenLiang** [comment]
Since now, NOT.
Our env variables were be set in script command, or in ci/build machine

**Comment by chenjiahan** [comment]
Variables prefixed with `MODERN_` will be exposed to the client code, see:

https://modernjs.dev/en/guides/basic-features/env-vars.html#using-environment-variables
